plugins/obs-filters/noise-suppress-filter.c

```diff
@@ -414,9 +460,40 @@ static void noise_suppress_update(void *data, obs_data_t *s)
 					ng->use_nvafx = false;
 				}
 			}
+			pthread_mutex_unlock(&ng->nvafx_mutex);
 		}
-		pthread_mutex_unlock(&ng->nvafx_mutex);
+		if ((strcmp(ng->fx, method) != 0)) {
+			ng->fx = bstrdup(method);
+			ng->intensity_ratio = intensity;
+			for (int i = 0; i < ng->channels; i++) {
+				/* Destroy previous FX */
+				pthread_mutex_lock(&ng->nvafx_mutex);
+				if (NvAFX_DestroyEffect(ng->handle[i]) !=
+				    NVAFX_STATUS_SUCCESS) {
+					do_log(LOG_ERROR,
+					       "FX failed to be destroyed.");
+					ng->use_nvafx = false;
+				} else
+					ng->handle[i] = NULL;
+				pthread_mutex_unlock(&ng->nvafx_mutex);
+				nvafx_initialize_internal(data);
+				pthread_mutex_unlock(&ng->nvafx_mutex);
```

Comment:
Why is the same mutex unlocked twice here?

Reply:
i goofed. 
Fixed in new push.